esp-hal-common/src/utils.rs

```diff
@@ -0,0 +1,232 @@
+//! Utils
```

Comment:
Not a huge fan of the "utils" name, could we rename this `reg` or `reg_access`, or something like that? @esp-rs/espressif feel free to suggest names too.

Reply:
I wasn't a big fan too of the name so I'll go with your suggestions.

Reply:
Done in the commit.